src/main/java/com/splunk/logging/HttpEventCollectorSender.java

```diff
@@ -248,6 +304,48 @@ private String serializeEventInfo(HttpEventCollectorEventInfo eventInfo) {
         return event.toString();
     }
 
+    /**
+     * Parses the event message on the basis of message format.
+     * 
+     * @param message the event message string
+     * @param format the event message format
+     * 
+     * @return parsed event message based on format
+     */
+    private Object parseEventMessage(final String message, final MessageFormat messageFormat) {
+        // if message is null or blank then return without parsing
+        if (message == null || message.trim().length() == 0) {
+            return message;
+        }
+
+        switch (messageFormat) {
+            case JSON:
```

Comment:
I would suggest to encapsulate this method within `enum MessageFormat`, so this will allow to get rid of `parseEventMessage` method and potentially allows adding new formats in future easier.

    enum MessageFormat {
        TEXT("text"),
        JSON("json") {
            @Override
            Object format(String message) {
                final Object jsonObject = JSONValue.parse(message);
                if (jsonObject == null) {
                    // If JSON parsing failed then it is likely a text message or a malformed JSON message. Return input message string in
                    // such an event.
                    return message;
                } else {
                    return jsonObject;
                }
            }
        }
        ...
       Object format(String message) {
            return message;
        }
and then in `HttpEventCollectorSender` simply:

    // Parse message on the basis of format
    final Object parsedMessage = this.messageFormat.format(eventInfo.getMessage());
    
     // event body
    JSONObject body = new JSONObject();